frontend/src/components/widget/WidgetRenderer.js

```diff
@@ -0,0 +1,629 @@
+      tabIndex,
+      autoFocus,
+      readonly,
+    };
+
+    switch (widgetType) {
+      case 'Date':
+        if (range) {
+          // TODO: Watch out! The datetimerange widget as exception,
+          // is non-controlled input! For further usage, needs
+          // upgrade.
+          return (
+            <DatetimeRange
+              {...dateRangeProps}
+              onChange={(value, valueTo) =>
+                onPatch(
+                  widgetField,
+                  value ? Moment(value).format(DATE_FORMAT) : null,
+                  null,
+                  valueTo ? Moment(valueTo).format(DATE_FORMAT) : null
+                )
+              }
+              field={widgetField}
+              timeZone={timeZone}
+            />
+          );
+        } else {
+          return (
+            <div className={this.getClassNames({ icon: true })}>
+              <DatePicker
+                {...dateProps}
+                timeFormat={false}
+                dateFormat={dateFormat || true}
+                value={widgetValue || widgetData[0].value}
+                patch={(date) =>
+                  onPatch(
+                    widgetField,
+                    this.generateMomentObj(date, DATE_FORMAT),
+                    null,
+                    null,
+                    true
+                  )
+                }
+                {...{
+                  isOpenDatePicker,
+                  timeZone,
+                }}
+              />
+            </div>
+          );
+        }
+      case 'ZonedDateTime':
+        return (
+          <div className={this.getClassNames({ icon: true })}>
+            <DatePicker
+              {...dateProps}
+              timeFormat={true}
+              dateFormat={dateFormat || true}
+              hasTimeZone={true}
+              value={widgetValue || widgetData[0].value}
+              patch={(date) =>
+                onPatch(
+                  widgetField,
+                  this.generateMomentObj(date, DATE_TIMEZONE_FORMAT),
+                  null,
+                  null,
+                  true
+                )
+              }
+              {...{
+                isOpenDatePicker,
+                timeZone,
+              }}
+            />
+          </div>
+        );
+      case 'Time':
+        return (
+          <div className={this.getClassNames({ icon: true })}>
+            <DatePicker
+              {...dateProps}
+              timeFormat={TIME_FORMAT}
+              dateFormat={false}
+              value={this.generateMomentObj(widgetValue, TIME_FORMAT)}
+              patch={(date) =>
+                onPatch(
+                  widgetField,
+                  this.generateMomentObj(date, TIME_FORMAT),
+                  null,
+                  null,
+                  true
+                )
+              }
+              tabIndex={tabIndex}
+            />
+          </div>
+        );
+      case 'Timestamp':
+        return (
```

Comment:
Here also and for others like as well. Would be nice to separate here to have <Timestamp {...props} /> and have TimeStamp defined in a separate component